### PR TITLE
inject: restore City of Khamoon mummy

### LINKED ISSUE #886: Feature Request: Add the extra mummy in City of Khamoon like the PS1 version
Is there any possibility you could add the extra mummy in the City of Khamoon the same as the PS1 version? Stella's Tomb Raider site describes this room as **"POOL & BLUE ROOM WITH MANY LEDGES"**

### PR DESCRIPTION
**Pull request by lahm86**
Resolves #886.

#### Checklist

- [x] I have read the [coding conventions](https://github.com/rr-/Tomb1Main/blob/master/CONTRIBUTING.md#coding-conventions)
- [x] I have added a changelog entry about what my pull request accomplishes, or it is an internal change

#### Description

This option places the mummy and its triggers as follows in City of Khamoon (triggered both in the flipped and unflipped rooms). So the mummy spawns by the switch, and is triggered on the tile below the pushblock.

image

I had to move `Stats_ObserveRoomsLoad` in `level.c` to take place post-injection for the enemy kill count to update correctly. While this wasn't a problem previously for the restored bat in Colosseum, it is here because we're changing the `index` on two sectors that don't currently have any floor data. So as a result, the kill count remains correct.

I've ticked off `fix floor data issues` in the readme for now as this is the final topic discussed in #755.

### CODE REVIEW

**Comment by lahm86** [comment]
@ajtudela Could you provide translations for this new option? Thanks.

**Review by rr- — COMMENTED** [review]
Can we make this option more generic, such as "restore PS1 enemies" and then only in the option description talk about the mummy in specific?

**Comment by lahm86** [comment]
> Can we make this option more generic, such as "restore PS1 enemies" and then only in the option description talk about the mummy in specific?

Updated now, and I updated the vars/enums as well to be consistent. LMK what you think.

image

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,7 @@
 - added the current music track and timestamp to the savegame so they now persist on load (#419)
 - added the triggered music tracks to the savegame so one shot tracks don't replay on load (#371)
 - added forward/backward input detection in line with TR2+ for jump-twists (#931)
+- added an option to restore the mummy in City of Khamoon room 25, similar to the PS version (#886)
```
Comment (walkawayy) [inline CHANGELOG.md]: Idc personally but all other changelog entries use `PS1`. Maybe we should be consistent? Or we could retroactively change them all to `PS` or `PSX`. Whatever @rr- prefers.

File: tools/config/Tomb1Main_ConfigTool/Resources/Lang/en.json [hunk 2]
```diff
@@ -239,6 +239,10 @@
       "Title": "Change Pierre spawn behavior",
       "Description": "Makes a freshly triggered (runaway) Pierre replace an already existing (runaway) Pierre."
     },
+    "restore_ps1_enemies": {
+      "Title": "Restore PS1 enemies",
+      "Description": "Adds the mummy that appears in the PlayStation version of City of Khamoon, room 25.\nNB Changing this option will require restarting the level."
```
Comment (walkawayy) [inline tools/config/Tomb1Main_ConfigTool/Resources/Lang/en.json]: What does `NB` mean?

**Review by walkawayy — CHANGES_REQUESTED** [review]

File: tools/config/Tomb1Main_ConfigTool/Resources/Lang/en.json (hunk 2 above)
Comment (lahm86) [reply tools/config/Tomb1Main_ConfigTool/Resources/Lang/en.json]: Nota bene - maybe too formal?

File: CHANGELOG.md (hunk 1 above)
Comment (rr-) [reply CHANGELOG.md]: I'm fine with both, props for consistency

File: tools/config/Tomb1Main_ConfigTool/Resources/Lang/en.json (hunk 2 above)
Comment (rr-) [reply tools/config/Tomb1Main_ConfigTool/Resources/Lang/en.json]: Let's just remove it, as it doesn't add much value

**Comment by walkawayy** [comment]
Also going to be annoying and link #352 since I believe that is also a floor data issue.

**Comment by lahm86** [comment]
> Also going to be annoying and link #352 since I believe that is also a floor data issue.

I'm not convinced this is a floor data issue. The trigger there has a timer value of 2, and changing this makes no difference to the outcome. I think there is a bug somewhere in the camera control, but haven't pinpointed it yet.

File: CHANGELOG.md (hunk 1 above)
Comment (lahm86) [reply CHANGELOG.md]: Changed to PS1 to match.

File: tools/config/Tomb1Main_ConfigTool/Resources/Lang/en.json (hunk 2 above)
Comment (lahm86) [reply tools/config/Tomb1Main_ConfigTool/Resources/Lang/en.json]: NB: NB removed :)

**Review by rr- — APPROVED** [review]

**Comment by ajtudela** [comment]
Done #953

**Review by walkawayy — APPROVED** [review]

**Pull request merged**